compiler/noirc_frontend/src/parser/parser.rs

```diff
@@ -191,6 +191,11 @@ fn module() -> impl NoirParser<ParsedModule> {
     })
 }
 
+/// This parser is used for parsing top level statements in macros
+pub fn top_level_item() -> impl NoirParser<TopLevelStatement> {
+    top_level_statement(module())
+}
```

Comment:
nit but it might be worth creating a module to put all these parsers which are to handle macros. This makes it easier for us to distinguish between parsers for user-code and generated code.